test/stmt/errors.swift

```diff
@@ -174,95 +174,79 @@ func thirteen() {
   }
 }
 
-// SR 6400
-
-enum SR_6400_E: Error {
-  case castError
-}
+// https://github.com/apple/swift/issues/48950
+protocol ClassProto: AnyObject {}
+do {
+  enum E: Error {
+    case castError
+  }
 
-struct SR_6400_S_1 {}
-struct SR_6400_S_2: Error {}
+  do {
+    struct S1 {}
+    struct S2: Error {}
 
-protocol SR_6400_FakeApplicationDelegate: AnyObject {}
-class SR_6400_FakeViewController {}
+    do {
+      throw E.castError
+    } catch is S1 {} // expected-warning {{cast from 'any Error' to unrelated type 'S1' always fails}}
 
-func sr_6400() throws {
-  do {
-    throw SR_6400_E.castError
-  } catch is SR_6400_S_1 { // expected-warning {{cast from 'any Error' to unrelated type 'SR_6400_S_1' always fails}}
-    print("Caught error")
-  }
-  
-  do {
-    throw SR_6400_E.castError
-  } catch is SR_6400_S_2 {
-    print("Caught error") // Ok
+    do {
+      throw E.castError
+    } catch is S2 {} // Ok
```

Comment:
These tests verify casting warnings and are not related to effects checking, which is why I inlined the throwing function as well.